src/DataConsumer.ts

```diff
@@ -316,9 +349,29 @@ export class DataConsumer extends EnhancedEventEmitter
 					break;
 				}
 
+				case 'bufferedamount':
+				{
+					const { bufferedAmount } = data as { bufferedAmount: number };
+
+					if (
+						bufferedAmount <= this._bufferedAmountLowThreshold &&
+						bufferedAmount <= this._bufferedAmount
+					)
```

Comment:
This code is wrong. It does not work. The condition must be:

- New buffered amount is less than threshold and current buffered amount is higher than threshold.

Also, please update `this._bufferedAmount` before emitting events:

```ts
const { bufferedAmount } = data as { bufferedAmount: number };
const previousBufferedAmount = this._bufferedAmount;

this._bufferedAmount = bufferedAmount;

if (previousBufferedAmount > this._bufferedAmountLowThreshold && bufferedAmount < this._bufferedAmountLowThreshold)
{
  this.safeEmit('bufferedamountlow', bufferedAmount);
}
```

Reply:
Also remove the 'sctpsendbufferfull' from the observer, please.

Reply:
alright

Reply:
This is marked as resolved, but `sctpsendbufferfull` is still emitted by the observer.

Reply:
ah, I didn't read the `sctpsendbufferfull` comment.

Reply:
This must also be removed from the docs: https://mediasoup.org/documentation/v3/mediasoup/api/#dataConsumer-observer-on-sctpsendbufferfull